src/PrestaShopBundle/Controller/Api/TranslationController.php

```diff
@@ -140,10 +140,24 @@ public function listTreeAction(Request $request)
                 throw new Exception('This \'selected\' param is not valid.');
             }
 
-            if ('modules' === $type) {
-                $tree = $this->getModulesTree($lang, $type, $selected, $search);
-            } else {
-                $tree = $this->getNormalTree($lang, $type, $selected, $search);
+            switch ($type) {
+                case 'themes':
+                    $tree = $this->getNormalTree($lang, $type, $selected, $search);
+                    break;
+
+                case 'modules':
+                    $tree = $this->getModulesTree($lang, $selected, $search);
+                    break;
+
+                case 'mails':
+                    // when emails body will be implemented, it should be a different type
+                    // because domain routes only support "type" & "selected/theme" as parameters
+                    $tree = $this->getMailsSubjectTree($lang, $search);
+                    break;
+
+                default:
```

Comment:
What else can be translated ?
If we expect nothing else to be translated, maybe we should throw an exception in the `default` ?

Reply:
(maybe not in a patch version though 😛)

Reply:
Yes there are other types, like `back` `others`

Reply:
^ What jo answered